File: where-is-my-medicine/apps/customer/create_icon.js
const fs = require('fs');
const zlib = require('zlib');

const w = 256, h = 256;
const ihdr = Buffer.alloc(13);
ihdr.writeUInt32BE(w, 0);
ihdr.writeUInt32BE(h, 4);
ihdr[8] = 8; // bit depth
ihdr[9] = 2; // color type RGB

const raw = Buffer.alloc((w * 3 + 1) * h);
for (let y = 0; y < h; y++) {
    const off = y * (w * 3 + 1);
    raw[off] = 0; // filter byte
    for (let x = 0; x < w; x++) {
        raw[off + 1 + x * 3] = 13;     // R - #0D
        raw[off + 1 + x * 3 + 1] = 148; // G - #94
        raw[off + 1 + x * 3 + 2] = 136; // B - #88
    }
}

const compressed = zlib.deflateSync(raw);

function chunk(type, data) {
    const buf = Buffer.concat([Buffer.from(type), data]);
    const crc = Buffer.alloc(4);
    const crcVal = zlib.crc32(buf);
    crc.writeInt32BE(crcVal);
    const len = Buffer.alloc(4);
    len.writeUInt32BE(data.length);
    return Buffer.concat([len, buf, crc]);
}

const sig = Buffer.from([137, 80, 78, 71, 13, 10, 26, 10]);
const png = Buffer.concat([
    sig,
    chunk('IHDR', ihdr),
    chunk('IDAT', compressed),
    chunk('IEND', Buffer.alloc(0))
]);

fs.writeFileSync('assets/icon.png', png);
console.log('Created assets/icon.png');

File: where-is-my-medicine/apps/customer/create_icon.py
import struct, zlib

def create_png(filename, width=256, height=256):
    # Create a simple teal colored PNG icon
    r, g, b = 13, 148, 136  # #0D9488 teal
    
    raw_data = b''
    for y in range(height):
        raw_data += b'\x00'  # filter byte
        for x in range(width):
            raw_data += struct.pack('BBB', r, g, b)
    
    def chunk(chunk_type, data):
        c = chunk_type + data
        crc = struct.pack('>I', zlib.crc32(c) & 0xffffffff)
        return struct.pack('>I', len(data)) + c + crc
    
    with open(filename, 'wb') as f:
        # PNG signature
        f.write(b'\x89PNG\r\n\x1a\n')
        # IHDR
        ihdr_data = struct.pack('>IIBBBBB', width, height, 8, 2, 0, 0, 0)
        f.write(chunk(b'IHDR', ihdr_data))
        # IDAT
        compressed = zlib.compress(raw_data)
        f.write(chunk(b'IDAT', compressed))
        # IEND
        f.write(chunk(b'IEND', b''))

create_png('assets/icon.png')
print('Created assets/icon.png')
